src/cljs/harja/tiedot/urakka/toteumat/velho_varusteet_tiedot.cljs

```diff
@@ -57,36 +20,68 @@
                   {:id 6 :nimi "Puuttuu" :css-luokka "kl-puuttuu"}
                   {:id 7 :nimi "Ei voitu tarkastaa" :css-luokka "kl-ei-voitu-tarkistaa"}])
 
-(defn hakuparametrit [{:keys [valinnat urakka]}]
-  (merge
-    (select-keys valinnat [:tie :aosa :aeta :losa :leta :hoitokauden-alkuvuosi :hoitovuoden-kuukausi :kuntoluokat :toteuma])
-    {:urakka-id (:id urakka)
-     :tietolajit (map varustetyyppi->tietolaji (:varustetyypit valinnat))}))
-
 (defn muodosta-tr-osoite [{:keys [tr-numero tr-alkuosa tr-alkuetaisyys tr-loppuosa tr-loppuetaisyys] :as rivi}]
   (if tr-loppuosa
     (str tr-numero "/" tr-alkuosa "/" tr-alkuetaisyys "/" tr-loppuosa "/" tr-loppuetaisyys)
     (str tr-numero "/" tr-alkuosa "/" tr-alkuetaisyys)))
 
+(defn tee-varustetyyppihaku [valinnat nimikkeisto]
+  (reify protokollat/Haku
+    (hae [_ teksti]
+      (go (let [varustetyypit (flatten (into [] (if (:kohdeluokat valinnat)
+                                                  (vals (select-keys nimikkeisto (:kohdeluokat valinnat)))
+                                                  (vals nimikkeisto))))
+
+                itemit (if (< (count teksti) 1)
+                         varustetyypit
+                         (filter #(not= (.indexOf (.toLowerCase (:otsikko %))
```

Comment:
Funktio .toLowerCase() heittää virheen jos :otsikko avainta ei ole olemassa eli jos sille annetaan nil arvo. En tiedä onko mahdollista että tämä voi puuttua, mutta jos kuitenkin katsotaan ensin että tuo avain olemassa, eli jotenkin näin?

```
(filter #(and
           (:otsikko %)
           (not= (.indexOf
                  (.toLowerCase (:otsikko %))
                  (.toLowerCase teksti)) -1))
  varustetyypit)
```
Tällöin koodin ajaminen jää ennen toLowerCasea

Reply:
Erittäin hyvä löytö, lisäsin nil-checkin.